### PR TITLE
Removed '-l' option for libs with absolute paths in generators (issue 6598).

### PR DESCRIPTION
**Pull request by DmitrySokolov**
Changelog: Feature: libraries provided with absolute paths are stay unchanged (without `-l` option)
Docs: [conan-io/docs#1635](https://github.com/conan-io/docs/pull/1635)

closes https://github.com/conan-io/conan/issues/6598

### CODE REVIEW

File: conans/tools.py [hunk 1]
```diff
@@ -221,6 +222,18 @@ def latest_vs_version_installed(*args, **kwargs):
     return tools_win.latest_vs_version_installed(output=_global_output, *args, **kwargs)
 
 
+def shell_quote(*args):
```
Comment (memsharded) [inline conans/tools.py]: Unless explicitly required, these things doesn't go to "tools" as they would get exposed in conanfile.py recipes. If needed in code, put that in the "generators" package. In any case, this seems a bit complex, not sure what problems it is addressing.

**Comment by DmitrySokolov** [comment]
The current behavior is:
"name" => `-lname`
"/name" => `-l/name`

Is it ok to change it? Someone can count on the current behavior.

**Comment by jgsogo** [comment]
I think that it is safe to check if it matches a full **absolute** path (the library should exist at that location). Can you think of any use-case of people using successfully things like `-l/usr/local/xxx` (paths that already exist)? 

---

PS.- editables will face some issues related to it, but I prefer to think about the workaround there than to overengineering here.

File: conans/tools.py (hunk 1 above)
Comment (DmitrySokolov) [reply conans/tools.py]: Shell paths must be quoted (escaped). `shlex.sh_quote()` just not working for Windows platform. What solution do you propose?

**Comment by DmitrySokolov** [comment]
What info should be in the changelog? (And what is the PR Description? Do you mean the first comment?)

**Comment by jgsogo** [comment]
Yes, the first comment. All of them have two lines there: 

image

Changelog could be something like: _"Feature: generators `xcode`, `pkg_config` and `qmake` won't prepend `-l` to the libraries if they are provided with the full path"_

And a good place for the docs could be in the `package_info` method (https://docs.conan.io/en/latest/reference/conanfile/methods.html#package-info)

Thanks!

**Comment by memsharded** [comment]
Hi @DmitrySokolov 

I have had a look at this. I see a potential issue, because:

> I know that there is -l:name option but Apple binutils does not understand it.

The problem is that according to https://stackoverflow.com/questions/50556472/linking-shared-library-absolute-vs-relative-path,

the syntax with absolute path only, works only if the shared library is using ``soname``.  But if it isn't it is necessary the ``-l:library`` syntax. I know that soname is most common, but what do you think of this? Thanks!

**Comment by DmitrySokolov** [comment]
I've just checked build commands generated by CMake on macOS, and all libraries are passed with absolute paths, even that uses custom names, i.e.

`-Wl,-rpath,/Users/...  /path/to/libName.dylib  /path/to/libName.a  /path/to/name.plugin`

**Comment by memsharded** [comment]
I am sorry this PR went very outdated, targeted to a very old Conan 1.X version in the ``develop`` branch, and there are some conflicts too.

I think it is better to close it, and if there is still interest in this, I'd suggest to target Conan 2.X in the ``develop2`` branch.
The very first step should be to open a PR with a failing test that illustrate the issue, and if possible just targeting one build system, ideally CMake, and if it only affects other build system like autotools, then autotools.

I apologize again for not handling this back then, often the load is too high and we cannot follow up on everything.

**Pull request closed without merging**